connectivity/check/features.go

```diff
@@ -0,0 +1,217 @@
+	Mode    string
+}
+
+func (s FeatureStatus) String() string {
+	str := "Disabled"
+	if s.Enabled {
+		str = "Enabled"
+	}
+
+	if len(s.Mode) == 0 {
+		return str
+	}
+
+	return fmt.Sprintf("%s:%s", str, s.Mode)
+}
+
+// FeatureSet contains the status
+type FeatureSet map[Feature]FeatureStatus
+
+// MatchRequirements returns true if the FeatureSet fs satisfies all the
+// requirements in reqs
+func (fs FeatureSet) MatchRequirements(reqs ...FeatureRequirement) bool {
+	for _, req := range reqs {
+		status := fs[req.feature]
+		if req.requiresEnabled && (req.enabled != status.Enabled) {
+			return false
+		}
+		if req.requiresMode && (req.mode != status.Mode) {
+			return false
+		}
+	}
+
+	// empty requirement list matches any feature set
+	return len(reqs) == 0
+}
+
+// FeatureRequirement defines a test requirement. A given FeatureSet may or
+// may not satisfy this requirement
+type FeatureRequirement struct {
+	feature Feature
+
+	requiresEnabled bool
+	enabled         bool
+
+	requiresMode bool
+	mode         string
+}
+
+// RequireFeatureEnabled constructs a FeatureRequirement which expects the
+// feature to be enabled
+func RequireFeatureEnabled(feature Feature) FeatureRequirement {
+	return FeatureRequirement{
+		feature:         feature,
+		requiresEnabled: true,
+		enabled:         true,
+	}
+}
+
+// RequireFeatureMode constructs a FeatureRequirement which expects the feature
+// to be in the given mode
+func RequireFeatureMode(feature Feature, mode string) FeatureRequirement {
+	return FeatureRequirement{
+		feature:      feature,
+		requiresMode: true,
+		mode:         mode,
+	}
+}
+
+func parseBoolStatus(s string) bool {
+	switch s {
+	case "Enabled", "enabled", "True", "true":
+		return true
+	}
+
+	return false
+}
+
+func (ct *ConnectivityTest) extractFeaturesFromConfigMap(ctx context.Context, client *k8s.Client, result FeatureSet) error {
+	cm, err := client.GetConfigMap(ctx, ct.params.CiliumNamespace, defaults.ConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to retrieve ConfigMap %q: %w", defaults.ConfigMapName, err)
+	}
+	if cm.Data == nil {
+		return fmt.Errorf("ConfigMap %q does not contain any configuration", defaults.ConfigMapName)
+	}
+
+	// Monitor aggregation level defaults to none.
+	mode := defaults.ConfigMapValueMonitorAggregatonNone
+	if v, ok := cm.Data[defaults.ConfigMapKeyMonitorAggregation]; ok {
+		mode = strings.ToLower(v)
+	}
+	result[FeatureMonitorAggregation] = FeatureStatus{
+		Enabled: mode != defaults.ConfigMapValueMonitorAggregatonNone,
+		Mode:    mode,
+	}
+
+	return nil
+}
+
+func (ct *ConnectivityTest) extractFeaturesFromCiliumStatus(ctx context.Context, ciliumPod Pod, result FeatureSet) error {
+	stdout, err := ciliumPod.K8sClient.ExecInPodWithTTY(ctx, ciliumPod.Pod.Namespace, ciliumPod.Pod.Name,
+		defaults.AgentContainerName, []string{"cilium", "status", "-o", "json"})
+	if err != nil {
+		return fmt.Errorf("failed to fetch cilium status: %w", err)
+
+	}
```

Comment:
nit: drop the empty line

```suggestion
		return fmt.Errorf("failed to fetch cilium status: %w", err)
	}
```